lxd/devices.go

```diff
@@ -484,6 +484,12 @@ func deviceTaskBalance(s *state.State) {
 			}
 		}
 
+		cpuPinStrategy := conf["limits.cpu.pin_strategy"]
```

Comment:
this validation shouldnt be here, it should be done where the rest of instance validation is checked so that invalid configs cant be stored.

Reply:
although is this needed because of the fact we do allow this key to be set in the profile?

Reply:
Sure thing!